### ISSUE TITLE
Unable to read parquet files with double nested float list 

### ISSUE THREAD

**Issue by Igosuki** [jorgecarleitao/arrow2#852]
I produced a simple parquet file using spark, I started monkey patching arrow2 to try and get it to read the file, but it doesn't seem to work.
Any idea ? 

The file : 
https://github.com/Igosuki/arrow2/blob/main/part-00000-b4749aa1-94e4-4ddb-bab2-954c4d3a290f.c000.snappy.parquet

The schema : 
```
schema = Schema { fields: [Field { name: "pr", data_type: Utf8, is_nullable: true, metadata: {} }, Field { name: "asks", data_type: List(Field { name: "element", data_type: List(Field { name: "element", data_type: Float64, is_nullable: true, metadata: {} }), is_nullable: true, metadata: {} }), is_nullable: true, metadata: {} }, Field { name: "bids", data_type: List(Field { name: "element", data_type: List(Field { name: "element", data_type: Float64, is_nullable: true, metadata: {} }), is_nullable: true, metadata: {} }), is_nullable: true, metadata: {} }, Field { name: "event_ms", data_type: Int64, is_nullable: true, metadata: {} }], metadata: {"org.apache.spark.sql.parquet.row.metadata": "{\"type\":\"struct\",\"fields\":[{\"name\":\"pr\",\"type\":\"string\",\"nullable\":true,\"metadata\":{}},{\"name\":\"asks\",\"type\":{\"type\":\"array\",\"elementType\":{\"type\":\"array\",\"elementType\":\"double\",\"containsNull\":true},\"containsNull\":true},\"nullable\":true,\"metadata\":{}},{\"name\":\"bids\",\"type\":{\"type\":\"array\",\"elementType\":{\"type\":\"array\",\"elementType\":\"double\",\"containsNull\":true},\"containsNull\":true},\"nullable\":true,\"metadata\":{}},{\"name\":\"event_ms\",\"type\":\"long\",\"nullable\":true,\"metadata\":{}}]}", "org.apache.spark.version": "3.2.0"} }
```

The patches https://github.com/Igosuki/arrow2/tree/tests_nesting

**Comment by jorgecarleitao** [comment]
Sorry, @Igosuki ; I am looking into this. It is related to reading dremel.

**Comment by Igosuki** [comment]
@jorgecarleitao Thank you, I was trying to debug it yesterday. Most of the arrow2 code actually works flawlessly and manages to read the file, until it considers that there are no more bytes to read despite having more rows since the decoding didn't infer the proper offsets (or something like that).

**Comment by jorgecarleitao** [comment]
I have a proposed fix here: #884.

Checking it via

```
cargo run --features io_parquet,io_parquet_compression --example parquet_read -- part-00000-b4749aa1-94e4-4ddb-bab2-954c4d3a290f.c000.snappy.parquet
```

**Comment by Igosuki** [comment]
Very cool, thank you !
